types/fetch-mock/index.d.ts

```diff
@@ -148,6 +149,16 @@ declare namespace fetchMock {
          */
         query?: { [key: string]: string };
 
+        /**
+         * key/value map of express style path params to match
+         */
+        params?: { [key: string]: string };
+
+        /**
+         * A function for arbitrary matching
+         */
+        functionMatcher?: MockMatcherFunction;
```

Comment:
Thanks for this! I think this should be a `MockMatcher` though to include the `string` and `RegExp` values that can be passed here:

```suggestion
        functionMatcher?: MockMatcher;
```

c.f. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d9ce66b18d497536f94b91ac02eb5b8d54ee33b7/types/fetch-mock/index.d.ts#L37

Reply:
@chrissinclair Thanks for reviewing, but `functionMatcher` should be a function. Would you please check below?

`getFunctionMatcher()` returns `functionMatcher` as is:
https://github.com/wheresrhys/fetch-mock/blob/edb007076e60fa0520b4463283f123222f62db58/src/lib/generate-matcher.js#L72

and it will be invoked at:
https://github.com/wheresrhys/fetch-mock/blob/edb007076e60fa0520b4463283f123222f62db58/src/lib/generate-matcher.js#L125

Reply:
Thanks @dobogo - I see what you're saying now. Get the feeling I was muddling this with `matcher` below, thanks for clarifying!